src/command.c

```diff
@@ -453,6 +456,60 @@ int cmd_proxy(struct command *cmd, struct redis_data *data)
 
     if (strcasecmp(type, "INFO") == 0) {
         return cmd_proxy_info(cmd);
+    } else if (strcasecmp(type, "SETREMOTES") == 0) {
+    	//lock global
+    	pthread_mutex_lock(&lock_setremotes);
+    	if (data->elements % 2 != 0 || data->elements < 4) {
+    		cmd_mark_fail(cmd, rep_addr_err);
+    	} else {
+    		int i = 0;
+    		char ip[45], port_s[10];
+    		struct address *addr = 0;
+    		int addr_cnt = 0;
+    		int addr_len = 0;
+    		int port_len = 0;
+    		for (i = 2; i < data->elements; i += 2) {
+    			long port = 0;
+    			addr_len = data->element[i].pos.str_len;
+    			port_len = data->element[i + 1].pos.str_len;
+				if (addr_len > sizeof(ip) - 1 || port_len > sizeof(port) - 1) {
+					continue;
+				}
+				if (pos_to_str(&data->element[i].pos, ip) == CORVUS_ERR) {
+					continue;
+				}
+				if (pos_to_str(&data->element[i + 1].pos, port_s) == CORVUS_ERR) {
+					continue;
+				}
```

Comment:
When the input is illegal, reject it with error message instead of just ignoring it.